Duplicati/Library/Common/IO/SystemIOWindows.cs

```diff
@@ -34,28 +34,39 @@ public struct SystemIOWindows : ISystemIO
         private const string PATHPREFIX_SERVER = @"\\";
         private static readonly string DIRSEP = Util.DirectorySeparatorString;
 
-        private static bool IsPathTooLong(string path)
-        {
-            // Use 258 for length check instead of 260 (MAX_PATH) - we need to leave room for the 16-bit (wide) null terminator
-            return path.StartsWith(UNCPREFIX, StringComparison.Ordinal) || path.StartsWith(UNCPREFIX_SERVER, StringComparison.Ordinal) || path.Length > 258;
-        }
-
         public static string PrefixWithUNC(string path)
         {
-            if (path.StartsWith(UNCPREFIX_SERVER, StringComparison.Ordinal))
-                return path;
-
-            if (path.StartsWith(UNCPREFIX, StringComparison.Ordinal))
+            if (IsPrefixedWithUNC(path))
+            {
                 return path;
-
+            }
             return path.StartsWith(PATHPREFIX_SERVER, StringComparison.Ordinal)
-                ? UNCPREFIX_SERVER + path.Remove(0, PATHPREFIX_SERVER.Length)
+                ? UNCPREFIX_SERVER + path.Substring(PATHPREFIX_SERVER.Length)
                 : UNCPREFIX + path;
         }
 
+        private static bool IsPrefixedWithUNC(string path)
+        {
+            return path.StartsWith(UNCPREFIX_SERVER, StringComparison.Ordinal) ||
+                path.StartsWith(UNCPREFIX, StringComparison.Ordinal);
+        }
+
         public static string StripUNCPrefix(string path)
         {
-            return path.StartsWith(UNCPREFIX, StringComparison.Ordinal) ? path.Substring(UNCPREFIX.Length) : path;
+            if (path.StartsWith(UNCPREFIX_SERVER))

```

Comment:
I think it would be better to specify `StringComparison.Ordinal` as the comparison type.

Reply:
Thanks.  I've pushed a fix.